app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,54 @@
+class ReservationsController < ApplicationController
+  before_action :authenticate_user!
+
+
+  def new
+    @reservation = Reservation.new
+    render "reservation", locals: {list_id: params[:id]}
+  end
+
+  # aka "update" method (there's two)
+  def ship
+    @reservation = Reservation.find(params[:id])
+
+    if @reservation.donor_id == current_user.id || current_user.administrator?
+      @reservation.ship
+      @reservation.save
```

Comment:
need to take this out, it's a remnant from my frustrations trying to get the specs to pass for this method